test/host.test.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
```

Comment:
Can you rebase to the latest `main` and place the test inside [request-header-host.test.js](https://github.com/fastify/fastify/blob/main/test/request-header-host.test.js)